src/packages/utils/src/things/json-rpc/json-rpc-quantity.ts

```diff
@@ -131,19 +134,98 @@ export class Quantity extends BaseJsonRpcType {
     return firstNonZeroByte;
   }
 
-  static toBuffer(value: JsonRpcInputArg, nullable?: boolean): Buffer {
+  private static getArgumentAsBigInt(argument: JsonRpcInputArg | Quantity): bigint {
+    const argumentType = typeof argument;
+    let argumentValue: bigint;
+
+    if (argumentType === "number") {
+      const argumentNumber = argument as number;
+      if (argumentNumber % 1) {
+        throw new Error("Cannot wrap a decimal as a json-rpc type.");
+      }
+      if (!isFinite(argumentNumber)) {
+        throw new Error(`Cannot wrap ${argument} as a json-rpc type.`);
+      }
+      argumentValue = BigInt(argumentNumber);
+    }  else if (argumentType === "bigint") {
+      argumentValue = argument as bigint;
+    } else if (argumentType === "string") {
+      if ((argument as string).slice(0, 2).toLowerCase() !== "0x") {
+        throw new Error(`Cannot wrap "${argument}" as a JSON-RPC type; strings must be hex-encoded and prefixed with "0x".`);
+      }
+      argumentValue = BigInt(argument as string);
+    } else if (argument instanceof Quantity) {
+      argumentValue = argument.toBigInt();
+    } else if (Buffer.isBuffer(argument)) {
+      argumentValue = bufferToBigInt(argument);
+    } else if (argumentValue == null) {
+      argumentValue = 0n;
+    }
+
+    return argumentValue;
+  }
+
+  public add(addend: JsonRpcInputArg | Quantity): Quantity {
+    if (this.bufferValue == null || this.bufferValue.length === 0) {
+      return addend instanceof Quantity ? addend : Quantity.from(addend, this._nullable);
+    }
+
+    const addendValue = Quantity.getArgumentAsBigInt(addend);
+
+    if (addendValue === undefined) {
+      throw new Error(`Cannot add ${addend} to a Quantity`);
+    }
+
+    const sum = this.toBigInt() + addendValue;
+
+    if (sum < 0n) {
+      throw new Error(`Cannot add ${addend} to a a Quantity of ${this}, as it results in a negative value`);
+    };
+
+    // convert to buffer directly, as this is cheaper than passing the bigint into the Quantity constructor
+    const sumBuffer = bigIntToBuffer(sum);
+    return new Quantity(sumBuffer, this._nullable);
+  }
+
+  public multiply(multiplier: JsonRpcInputArg | Quantity): Quantity {
+    if (this.bufferValue == null ||this.bufferValue.length === 0) {
+      return this._nullable ? Quantity.Empty : Quantity.Zero;
+    }
+
+    const multiplierValue = Quantity.getArgumentAsBigInt(multiplier);
+    if (multiplierValue < 0) {
+      throw new Error(`Cannot multiply a Quantity by a negative multiplier`);
+    }
+
+    if (multiplierValue === undefined) {
+      throw new Error(`Cannot multiply a Quantity by ${multiplier}`);
+    }
+
+    const thisValue = this.toBigInt();
+
+    const productBuffer = bigIntToBuffer(thisValue * multiplierValue);
+    return new Quantity(productBuffer, this._nullable);
+  }
+
+  static toBuffer(value: JsonRpcInputArg): Buffer;
+  static toBuffer(value: JsonRpcInputArg, nullable?: boolean): Buffer | null {
```

Comment:
I added overrides to the `toXXXX()` methods so that calling the function without supplying `nullable` (defaults to `false`), returns a non-nullable type. It'd be nice to do this on the instance methods too, but I think that will be a more significant undertaking.